code/modules/mining/equipment/mining_charges.dm

```diff
@@ -0,0 +1,215 @@
+	det_time = 5
+	notify_admins = FALSE // no need to make adminlogs on lavaland, while they are "safe" to use
+	var/timer_off = FALSE // when TRUE, charges won't detonate on it's own. Used for mining detonator
+	var/installed = FALSE
+	var/smoke_amount = 3
+	/// list of sizes for explosion. Third number is used for actual rock explosion size, second number is radius for Weaken() effects, first is used for hacked charges
+	var/boom_sizes = list(2, 3, 5)
+	var/hacked = FALSE
+
+/obj/item/grenade/plastic/miningcharge/Initialize()
+	. = ..()
+	image_overlay = mutable_appearance(icon, "[icon_state]_active", ON_EDGED_TURF_LAYER)
+
+/obj/item/grenade/plastic/miningcharge/examine(mob/user)
+	. = ..()
+	if(hacked)
+		. += "Its wiring is haphazardly changed."
+	if(timer_off)
+		. += "<span class='notice'>The mining charge is connected to a detonator.</span>"
+
+/obj/item/grenade/plastic/miningcharge/attack_self(mob/user)
+	if(nadeassembly)
+		nadeassembly.attack_self(user)
+
+/obj/item/grenade/plastic/miningcharge/afterattack(atom/movable/AM, mob/user, flag)
+	if(!ismineralturf(AM) && !hacked)
+		return
+	if(is_ancient_rock(AM) && !hacked)
+		visible_message("<span class='notice'>This rock appears to be resistant to all mining tools except pickaxes!</span>")
+		return
+	if(!timer_off) //override original proc for plastic explosions
+		return ..()
+	if(!flag)
+		return
+	if(iscarbon(AM))
+		return
+	to_chat(user, "<span class='notice'>You start planting the [src].</span>")
+	if(do_after(user, 25 * toolspeed, target = AM))
+		if(!user.unEquip(src))
+			return
+		target = AM
+		loc = null
+		if(hacked)
+			message_admins("[ADMIN_LOOKUPFLW(user)] planted [name] on [target.name] at [ADMIN_COORDJMP(target)]")
+			log_game("planted [name] on [target.name] at [COORD(target)]", user)
+		installed = TRUE
+		target.overlays += image_overlay
+
+/obj/item/grenade/plastic/miningcharge/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/detonator))
+		return
+	var/obj/item/detonator/detonator = I
+	if(!(src in detonator.bombs) && !timer_off)
+		detonator.bombs += src
+		timer_off = TRUE
+		to_chat(user, "<span class='notice'>You synchronized [src] to a detonator.</span>")
+		playsound(src, 'sound/machines/twobeep.ogg', 50)
+		detonator.update_icon()
+	else
+		to_chat(user, "<span class='warning'>[src] was already synchronized to a existing detonator!</span>")
+
+/obj/item/grenade/plastic/miningcharge/proc/detonate()
+	addtimer(CALLBACK(src, PROC_REF(prime)), 3 SECONDS)
+
+/obj/item/grenade/plastic/miningcharge/prime()
+	if(hacked) //try not to blow your fingers off
+		explode()
+		return
+	var/turf/simulated/mineral/location = get_turf(target)
+	var/datum/effect_system/smoke_spread/S = new
+	S.set_up(smoke_amount, 0, location, null)
+	S.start()
+	//location.attempt_drill(null,TRUE,3) //Ru-paradise uses special hardness system - some rocks can be destroyed only after 2 or 3 hits. It stays here as the reminder that ss220 lavaland is better.
```

Comment:
Don't really need this commented out code here, feel free to propose the hardness system if you want to, since I didn't even know it was a thing downstream
```suggestion
```